### PR TITLE
Reset Password via email

### PR DESCRIPTION
**Pull request by jamesndekezi**
#### What does this PR do?

Reset Password via email

#### Describe your changes

This PR adds the functionality to reset a user's password via email. When a user forgets their password, they can request a reset, and an email with a reset link will be sent to them. This link allows the user to reset their password securely.

#### PR Type
What kind of change does this PR introduce?
```
[ ] Bugfix
[x] Feature
[ ] Refactoring (no functional changes, no API changes)
[ ] Documentation content changes
[ ] Tests
[ ] Other


#### How should this be manually tested?

Go to the login page and click on "Forgot Password".
Enter the registered email address.
Check the email inbox for a reset link.
Click on the reset link and set a new password.
Verify that the new password works for logging into the application.

#### What are the relevant pivotal tracker stories?
[Pivotal Tracker Link](https://xsolve.atlassian.net/browse/PROJECT_KEY)

#### Screenshots (if is appropriate)
[ ] Feature
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Tests
[ ] Other
```

#### How should this be manually tested?

Navigate to the login page and click on "Forgot Password".
Enter the registered email address.
Check the email inbox for a reset link.
Click on the reset link and set a new password.
Verify that the new password works for logging into the application.

#### What are the relevant pivotal tracker stories?
[Pivotal Tracker Link](https://xsolve.atlassian.net/browse/PROJECT_KEY)

#### Screenshots (if is approp
ffw
riate)

### CODE REVIEW

**Review by GabrielDushime — DISMISSED** [review]

**Review by mugishaj092 — CHANGES_REQUESTED** [review]
Your code is not working

File: eslint.config.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+const eslint = require("@eslint/js");
+const tseslint = require("typescript-eslint");
```
Comment (mugishaj092) [inline eslint.config.js:2]: This eslint.config.json is not necessary

File: .eslintrc.json [hunk 2]
```diff
@@ -1,41 +0,0 @@
-{
-    "root": true,
```
Comment (mugishaj092) [inline .eslintrc.json:2]: Why did you delete this file?

you need to restore it because is neccessary

File: src/middlewares/isUserFound.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { NextFunction, Request, Response } from "express";
```
Comment (mugishaj092) [inline src/middlewares/isUserFound.tsx:1]: This is not middleware you can put it in utils

File: src/utils/email.utils.ts [hunk 4]
```diff
@@ -3,14 +3,16 @@ import dotenv from "dotenv";
 dotenv.config();
```
Comment (mugishaj092) [inline src/utils/email.utils.ts:4]: This changes was not neccessary

**Pull request closed without merging**